### PR TITLE
[2198] Containerize Nginx for E2E test for local development

### PR DESCRIPTION
**Pull request by JSkeets**
## Related Issue or Background Info
https://github.com/CDCgov/prime-simplereport/issues/2198

Currently, as a developer, if the end to end test are ran locally the developer is required to set up their own reverse proxy for the test to successfully run. This pull request simplifies that process and creates a reverse proxy for the developer quickly and seamlessly.

## Changes Proposed

- Create NGINX dockerfile
- Integrate launching of Nginx container with script to run end to end test

## Additional Information

- A goal of the ticket was to prevent developers from having to modify their `/etc/hosts` file. The current pull request does not address this issue, the `/etc/hosts` file must be modified and the test have to be rendered from a domain other than localhost due to the following constraint [here](https://stackoverflow.com/questions/1134290/cookies-on-localhost-with-explicit-domain/1188145#1188145)

## Screenshots / Demos
Successful launching of the container, and running of the test with existing `npm run e2e` command

Screen Shot 2021-08-11 at 1 37 26 PM

## Checklist for Author and Reviewer
Checkout code run `npm run e2e` from `frontend` directory

### Testing
- [ ] Includes a summary of what a code reviewer should verify

### CODE REVIEW

**Review by timothybest-usds — COMMENTED** [review]
.

**Comment by nickclyde** [comment]
This looks good! To address the `/etc/hosts` issue, we could run the e2e tests from inside a docker container with a modified hosts file. Should we include that here or spin it into a separate ticket?

File: frontend/e2e/utils/nginx/conf [hunk 1]
```diff
@@ -0,0 +1,21 @@
+server {
```
Comment (jdorothy) [inline frontend/e2e/utils/nginx/conf:1]: General comment - I think I'd expect a file to named `something.conf` rather than just `conf`

**Review by jdorothy — APPROVED** [review]

**Comment by jdorothy** [comment]
> This looks good! To address the `/etc/hosts` issue, we could run the e2e tests from inside a docker container with a modified hosts file. Should we include that here or spin it into a separate ticket?

I agree - I think the idea is to not modify the local filesystem at all, so we can leverage Docker networking between containers to accomplish that. In terms of big picture, we can use this to run the application stack locally as well and present an nginx container that mimics the cloud environment's load balancer. That way tests and the local app are run in the same way using the same container.

**Pull request merged**